src/chain/mod.rs

```diff
@@ -73,3 +83,117 @@ pub async fn export<D: Digest>(
 
     Ok(digest)
 }
+
+pub async fn export_v2<D: Digest>(
```

Comment:
I know we don't have unit tests tests for the original export, but it'd be great to have some for `v2`.